src/@types/Compute.ts

```diff
@@ -0,0 +1,53 @@
+import { Metadata, MetadataAlgorithm } from './Metadata'
+
+export type ComputeResultType = 'algorithmLog' | 'output'
+
+export interface ComputeResult {
+  filename: string
+  filesize: number
+  type: ComputeResultType
+  index: number
```

Comment:
please make index optional
```suggestion
  index?: number
```